lib/ice_cube/rules/weekly_rule.rb

```diff
@@ -35,7 +35,17 @@ def realign(step_time, start_time)
       time = TimeUtil::TimeWrapper.new(start_time)
       offset = wday_offset(step_time, start_time)
       time.add(:day, offset)
-      super(step_time, time.to_time)
+      realigned_time = time.to_time
+      # when the realigned time is in a different hour, it means that
+      # time falls to the DST switch timespan. In this case, we need to
+      # move the time back by one day to ensure that the hour stays the same
+      # WARNING: this could not work if the DST change is on a monday
+      # as the realigned time would be moved to the previous week.
+      if realigned_time.hour != start_time.hour
+        time.add(:day, -1)
+        realigned_time = time.to_time
+      end
```

Comment:
Can we not just detect the difference between the `start_time.hour` and the `realigned_time.hour` and make a 60 minute adjustment to the realigned_time to correct for any DST changes? That would avoid any issues with switching weeks, etc.

Reply:
This case can only happen when the realigned time changed the hour in the internals of `Time` because of a DST change, so -1 day will always be a time which does not fall into the DST change time imho.

I also see that this is not the ideal solution, it would be better to somehow "fix" the hour to the start time hour of the schedule somewhere in `ValidatedRule#next_time` but I did not find a way to access the start time of the schedule in `ValidatedRule`. Maybe you have an idea to that approach.

Regarding the edge case: that was just a guess based on the documentation for the message. I don't really understand the described edge case and thought that the movement to the previous day could somehow trigger that case again.